be/src/vec/exec/format/json/new_json_reader.cpp

```diff
@@ -1370,53 +1374,40 @@ Status NewJsonReader::_simdjson_handle_nested_complex_json(
     return Status::OK();
 }
 
-size_t NewJsonReader::_column_index(const StringRef& name, size_t key_index) {
-    /// Optimization by caching the order of fields (which is almost always the same)
-    /// and a quick check to match the next expected field, instead of searching the hash table.
-    if (_prev_positions.size() > key_index && _prev_positions[key_index] &&
-        name == _prev_positions[key_index]->get_first()) {
-        return _prev_positions[key_index]->get_second();
-    } else {
-        auto* it = _slot_desc_index.find(name);
-        if (it) {
-            if (key_index < _prev_positions.size()) {
-                _prev_positions[key_index] = it;
-            }
-            return it->get_second();
-        } else {
-            return size_t(-1);
-        }
-    }
-}
-
 Status NewJsonReader::_simdjson_set_column_value(simdjson::ondemand::object* value, Block& block,
                                                  const std::vector<SlotDescriptor*>& slot_descs,
                                                  bool* valid) {
     // set
     _seen_columns.assign(block.columns(), false);
     size_t cur_row_count = block.rows();
     bool has_valid_value = false;
-    // iterate through object, simdjson::ondemond will parsing on the fly
-    size_t key_index = 0;
-    for (auto field : *value) {
-        std::string_view key = field.unescaped_key();
-        StringRef name_ref(key.data(), key.size());
-        const size_t column_index = _column_index(name_ref, key_index++);
-        if (UNLIKELY(ssize_t(column_index) < 0)) {
-            // This key is not exist in slot desc, just ignore
+    for (size_t i = 0; i < slot_descs.size(); ++i) {
+        auto slot_desc = slot_descs[i];
+        if (!slot_desc->is_materialized()) {
             continue;
         }
-        simdjson::ondemand::value val = field.value();
-        auto* column_ptr = block.get_by_position(column_index).column->assume_mutable().get();
-        RETURN_IF_ERROR(
-                _simdjson_write_data_to_column(val, slot_descs[column_index], column_ptr, valid));
-        if (!(*valid)) {
-            return Status::OK();
+        auto* column_ptr = block.get_by_position(i).column->assume_mutable().get();
+        auto field = value->find_field_unordered(slot_desc->col_name());
```

Comment:
this is much slower than iterate through `for (auto field : *value) `, this loop will utilize simd instruction

Reply:
Thanks, I will fix it